doc/web/source/include/PluginForRavenCode.rst

```diff
@@ -0,0 +1,195 @@
+
+Test Location and Definition
+----------------------------
+
+LOGOS is a RAVEN-supported plugin and is managed by Git. RAVEN plugins
+provide an option to associate a workflow or a set of RAVEN external
+models with RAVEN without including them directly in the main RAVEN
+repository. The benefits include modularity, access restriction, and
+regression testing for compatibility with RAVEN as it continues to
+evolve. In this way, we are able to treat this repository as separate,
+yet still use one from within the other.
+
+.. The main structure of the LOGOS repository is shown in
+.. Figure :ref:`fig-Logos`.
+
+The folder *LOGOS/tests* contains the inputs for the
+tests, in addition to the corresponding output (in the ``gold`` folder)
+used to ensure that the behavior of the code does not change when new
+modifications are introduced. These tests can also be organized in
+subfolders based on their characteristics.
+
+.. .. _fig-Logos:
+.. .. figure:: ../figures/LogosRepo.jpg
+..    :scale: 50%
+..    :align: center
+
+..    Folder tree of the LOGOS repository.
+
+
+The RAVEN repository contains a complete testing system for providing
+regression testing for itself and for its plugins. The LOGOS module
+tests are defined in the same manner as those for RAVEN itself. A single
+test consists of a RAVEN input file along with the associated data
+needed to perform that run. This can include input data, external
+models, and Python files. These may be placed in the *tests* directory
+or in related subdirectories.
+
+Each directory that contains tests to be run by the framework must
+contain a test specification file named ``tests``. The syntax of these
+files is defined by the RAVEN test framework, which controls how each
+test is run and sets the criteria used to determine whether it passed or
+not. An example of a test specification file is:
+
+.. code-block:: python
+
+   [Tests]
+     [./skp_optimization]
+       type = 'RavenFramework'
+       input = 'test_skp.xml'
+       UnorderedCsv = 'skp/test_skp.csv'
+     [../]
+   []
+
+In the example above, one test, named ``skp_optimization``, is defined
+using the RAVEN test module (``type = 'RavenFramework'``). Comparison
+criteria are also defined in the ``tests`` file. In most cases, one or
+more output files generated by running the specified input file with
+RAVEN are compared against a gold standard provided by the developer and
+stored in the repository. Typically, comparisons are performed on
+numeric values contained in CSV files up to a defined tolerance.
+
+When these file comparisons are specified by the test developer,
+reference files must have the same name and be placed in the ``gold``
+subdirectory below the directory containing the ``tests`` file.
+
+Running the Tests
+-----------------
+
+The integrated tests can also be run manually, for example by executing:
+
+.. code-block:: bash
+
+   path/to/raven/raven_framework path/to/LOGOS/tests/TestName
+
```

Comment:
I would add indication to run unit tests as well

Reply:
I would suggest to update it later. I prefer to use Pytest for unit testing. Let's wait for our future development